operator/v1/types_ingress.go

```diff
@@ -925,6 +936,32 @@ type IngressControllerHTTPHeaders struct {
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
 }
 
+// IngressControllerHTTPHeaderBuffer specifies the size of the
+// per-connection HTTP header buffers.
+type IngressControllerHTTPHeaderBuffer struct {
+	// headerBufferBytes describes how much memory should be reserved
+	// (in bytes) for IngressController connection sessions.
+	// Note that this value must be at least 16384 if HTTP/2 is
+	// enabled for the IngressController (https://tools.ietf.org/html/rfc7540).
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=16384
```

Comment:
Sorry, one last thing:  I believe you need to specify a maximum value as well for both fields.

Reply:
It's not clear to me if a maximum value is required in addition to a minimum value:

https://github.com/openshift/api/blob/master/operatoringress/v1/types.go#L52

Im assuming the maximum would be defaulted to MAXINT anyways, right?

Reply:
If we _want_ to include an explicit maximum, we can, but im not sure what value would make sense.

Reply:
I'd want to check with @deads2k if that is sufficient.

Reply:
For an integer, you should be as restrictive as is reasonable for your API.  For slices, many APIs have a practical upper bound before they won't work very well.  This is especially true for configuration APIs, where there is often fanout associated with maintenance of items, so on the recent Route{Spec,Status} I asked for upper bounds on several fields.  

For an integer like this, the max size of the item doesn't cause fanout issues, so I wouldn't worry about it.